pkg/clients/iam/iamrole.go

```diff
@@ -1,18 +1,138 @@
 package iam
 
 import (
+	"net/url"
+	"strings"
+
 	"github.com/aws/aws-sdk-go-v2/aws"
 	"github.com/aws/aws-sdk-go-v2/service/iam"
+	"github.com/google/go-cmp/cmp"
+	"github.com/google/go-cmp/cmp/cmpopts"
+	"github.com/mitchellh/copystructure"
+	"github.com/pkg/errors"
+
+	"github.com/crossplane/provider-aws/apis/identity/v1beta1"
+	awsclients "github.com/crossplane/provider-aws/pkg/clients"
 )
 
+const errCheckUpToDate = "unable to determine if external resource is up to date"
+
 // RoleClient is the external client used for IAMRole Custom Resource
 type RoleClient interface {
 	GetRoleRequest(*iam.GetRoleInput) iam.GetRoleRequest
 	CreateRoleRequest(*iam.CreateRoleInput) iam.CreateRoleRequest
 	DeleteRoleRequest(*iam.DeleteRoleInput) iam.DeleteRoleRequest
+	UpdateRoleRequest(*iam.UpdateRoleInput) iam.UpdateRoleRequest
 }
 
 // NewRoleClient returns a new client using AWS credentials as JSON encoded data.
 func NewRoleClient(conf *aws.Config) (RoleClient, error) {
 	return iam.New(*conf), nil
 }
```

Comment:
Ideally we would support multiple auth methods here, as you can see demonstrated here: https://github.com/crossplane/provider-aws/blob/1d51592300403eea476170ac4f1b549cf1ae42a6/pkg/clients/rds/rds.go#L48

Reply:
See comment on the controller for more context. I would be okay with leaving this as-is for now if needed.

Reply:
Going to go ahead and resolve this and will see about addressing in a later PR :)